### ISSUE TITLE
Turn menu items on/off at run-time

### ISSUE THREAD

**Issue by HiromuHota** [HiromuHota/pentaho-kettle#96]
Requested by a user company, menu items such as File > Save as... should be customizable.
"Customizable" means each one of the items can be turned on/off individually at run-time by e.g., a config file.

![screen shot 2018-02-09 at 10 16 02 am](https://user-images.githubusercontent.com/20668349/36042899-48122ec6-0d82-11e8-936f-515eb77febe8.png)

Example config file:
```
<item name="File">
  <item name="Save as...">
    <enable>false</enable>
  </item>
</item>
```

**Comment by delfer** [comment]
Good idea! I think it will be better to make it based on user's roles and rights. For example, integrate with Spring Security.

**Comment by HiromuHota** [comment]
That's an even better idea!

I'm not sure, but it would be an good idea to be able to turn them on/off in `ui/menubar.xul` like

```
<menuitem id="file-save-as" label="${Spoon.Menu.File.SaveAs}" command="spoon.saveFileAs()" disabled="spoon.checkMenuPermission(\"file-save-as\")" />
```

Then `spoon.checkMenuPermission(String id)` will check if the menu item should be disabled based on user's roles.

**Comment by HiromuHota** [comment]
I found that even as of today menu items can be hidden by commenting out (or completely deleting) the corresponding XML element in `ui/menubar.xul`.
For example, commenting out `file-save-as-vfs`

```
<menu id="file">
  ...
  <!--<menuitem id="file-save-as-vfs" label="${Spoon.Menu.File.SaveAsVFS}" command="spoon.saveXMLFileToVfs()" />-->
  ...
</menu>
```

will give you a menu without the commented menuitem as below.

![screen shot 2018-03-01 at 10 03 04 am](https://user-images.githubusercontent.com/20668349/36861056-c4c84562-1d37-11e8-87d6-884a7e9f7684.png)

**Comment by HiromuHota** [comment]
4d54235c60d9c21c89c0175be16f084aa2c58ceb supports attribute "disabled" to turn menuitems on/off.
The following `ui/menubar.xul`
```
<menu id="file">
  ...
  <menuitem id="file-save-as" label="${Spoon.Menu.File.SaveAs}" command="spoon.saveFileAs()" disabled="true" />
  <!--<menuitem id="file-save-as-vfs" label="${Spoon.Menu.File.SaveAsVFS}" command="spoon.saveXMLFileToVfs()" />-->
  ...
</menu>
```
will disable `file-save-as` and hide `file-save-as-vfs`.
![screen shot 2018-03-01 at 11 35 53 am](https://user-images.githubusercontent.com/20668349/36865577-bdc1a13e-1d44-11e8-87c7-4424ee15e9a5.png)

**Comment by HiromuHota** [comment]
Although Pentaho's XUL extends Mozilla's XUL (see [here](https://wiki.pentaho.com/display/ServerDoc2x/03.+XUL+1.0+Extensions+and+the+Pentaho+XUL+Namespace) for details), it does not seem to support enable/disable items based on role/user.
The XUL spec should be further extended to support it, which would need more holistic design hence more time.
Thoughts, design/code contributions will be welcome.

**Comment by yuanxuzhang** [comment]
> Requested by a user company, menu items such as File > Save as... should be customizable.
> "Customizable" means each one of the items can be turned on/off individually at run-time by e.g., a config file.
> 
> ![screen shot 2018-02-09 at 10 16 02 am](https://user-images.githubusercontent.com/20668349/36042899-48122ec6-0d82-11e8-936f-515eb77febe8.png)
> 
> Example config file:
> 
> ```
> <menu id="file">
>   <menuitem id="file-save-as" disabled="true" />
> </menu>
> ```

is there any way to hidden or remove menubar ?

**Comment by HiromuHota** [comment]
Currently, it is not possible to hide the entire menubar (File, Edit, View, Action, etc.).
Can you tell us why you want to hide it?

**Comment by yuanxuzhang** [comment]
I want to embed webspoon in a iframe. It satisfies the needs without menubar,and what's more, the whole page layout looks better.
